### ISSUE TITLE
events made easy and tinymce

### ISSUE THREAD

**Issue by herrvigg** [herrvigg/qtranslate-x#152]
<a href="https://github.com/liedekef"><img src="https://avatars0.githubusercontent.com/u/75667?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [liedekef](https://github.com/liedekef)**
_Tuesday May 05, 2015 at 22:52 GMT_
_Originally opened as https://github.com/qTranslate-Team/qtranslate-x/issues/152_

----

Hi,

the plugin is working fine with the combo qtranslate-x and EME, but there seems to be an issue with tinymce:
https://wordpress.org/support/topic/bug-when-switch-languages-on-visual-editor
I tried it on my local install and it seems to be the case for me too. Although in the generated html output I see some qtranxs-translatable class being added to the content div, the language switch only works for the text-mode. Maybe the sequence of loading the javascript is wrong (just a thought)?

Franky

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Thursday May 07, 2015 at 00:13 GMT_

----

@liedekef Franky: this is indeed strange, it all looks very similar to standard post page, but somehow it does not work. Can you think of a difference why event page could behave differently? Do you do anything non-wp-standard to load TinyMCE? I believe, if everything is done in the same way as on a standard post page '/wp-admin/post.php', it will work.

**Comment by herrvigg** [comment]
<a href="https://github.com/liedekef"><img src="https://avatars0.githubusercontent.com/u/75667?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [liedekef](https://github.com/liedekef)**
_Thursday May 07, 2015 at 10:14 GMT_

----

I haven't changed the code to add tinymce in quite a while. I use the standard wp call wp_editor in eme_events.php:

```
wp_editor($event['event_notes'],"content");
```

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Saturday May 16, 2015 at 17:54 GMT_

----

Hi Franky, would you have time to try the latest beta  https://github.com/qTranslate-Team/qtranslate-x/releases/tag/3.3.2 ? I think this issue has disappeared, could you confirm?

**Comment by herrvigg** [comment]
<a href="https://github.com/liedekef"><img src="https://avatars0.githubusercontent.com/u/75667?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [liedekef](https://github.com/liedekef)**
_Saturday May 16, 2015 at 23:00 GMT_

----

That version seems to be working. May I ask what it is that is special for the EME tinymce editor?

**Comment by herrvigg** [comment]
<a href="https://github.com/johnclause"><img src="https://avatars2.githubusercontent.com/u/5491361?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [johnclause](https://github.com/johnclause)**
_Sunday May 17, 2015 at 02:47 GMT_

----

I did not specifically worked on this issue, decided to take a look after the last round of changes to implement Integration Framework, and when I tried eme again, it worked. It is kind of hard to trace back now what was wrong, and maybe it is not important at this point, the only thing that comes in mind is that I moved line "qtx.addContentHooksTinyMCE();" to be executed always and not only on post.php. Ah, that is it, yes, you use admin.php to edit events, so that line was not fired for you before, which means it never worked, we missed it ;) Yes, normally, plugins would use post.php for any kind of posts, and "event" is one of them, but I guess you implemented your own framework.

I am closing it now, we can still write to a closed issue or may reopen it if needed.
